### ISSUE TITLE
StackDriver Logging python exception in protobuf json_format.py

### ISSUE THREAD

**Issue by mishedone** [googleapis/google-cloud-python#6217]
Scenario: I am trying to setup logging in Python3 AppEngine standard environment. The following code is executing without any problem:

```
from google.cloud import logging
client = logging.Client()
client.setup_logging()
```

But... When I try to log a warning StackDriver shows a Failed to submit 1 logs. error:

```
import logging
logger = logging.getLogger('records')
logger.warning('First try.')
```

Here is the stack trace from StackDriver (actually 3):

```
Traceback (most recent call last):
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 517, in _ConvertFieldValuePair self._ConvertMapFieldValue(value, message, field)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 654, in _ConvertMapFieldValue value[key], value_field)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 683, in _ConvertScalarFieldValue if _UNPAIRED_SURROGATE_PATTERN.search(value): TypeError: expected string or bytes-like object
```
```
Traceback (most recent call last):
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 547, in _ConvertFieldValuePair self.ConvertMessage(value, sub_message)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 452, in ConvertMessage self._ConvertFieldValuePair(value, message) 
File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 558, in _ConvertFieldValuePair raise ParseError('Failed to parse {0} field: {1}.'.format(name, e)) google.protobuf.json_format.ParseError: Failed to parse labels field: expected string or bytes-like object.
```

```
Traceback (most recent call last):
 File "/env/lib/python3.7/site-packages/google/cloud/logging/handlers/transports/background_thread.py", line 118, in _safely_commit_batch batch.commit()
 File "/env/lib/python3.7/site-packages/google/cloud/logging/logger.py", line 612, in commit client.logging_api.write_entries(entries, **kwargs)
 File "/env/lib/python3.7/site-packages/google/cloud/logging/_gapic.py", line 121, in write_entries entry_pbs = [_log_entry_mapping_to_pb(entry) for entry in entries]
 File "/env/lib/python3.7/site-packages/google/cloud/logging/_gapic.py", line 121, in <listcomp> entry_pbs = [_log_entry_mapping_to_pb(entry) for entry in entries]
 File "/env/lib/python3.7/site-packages/google/cloud/logging/_gapic.py", line 461, in _log_entry_mapping_to_pb ParseDict(mapping, entry_pb)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 421, in ParseDict parser.ConvertMessage(js_dict, message)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 452, in ConvertMessage self._ConvertFieldValuePair(value, message)
 File "/env/lib/python3.7/site-packages/google/protobuf/json_format.py", line 552, in _ConvertFieldValuePair raise ParseError('Failed to parse {0} field: {1}'.format(name, e)) google.protobuf.json_format.ParseError: Failed to parse resource field: Failed to parse labels field: expected string or bytes-like object.
```

Those look extremely weird - I am not passing any labels at all, it is using the default ones...

Here is the content of my requirements.txt

```
Django==2.1.1
djangorestframework==3.8.2
google-cloud-logging==1.7.0
google-cloud-pubsub==0.38.0
psycopg2-binary==2.7.5
```

I am running in a debian/stretch64 vagrant provisioned VM BUT the error is triggered in production.

**Comment by tseaver** [comment]
@mishedone Can you please show the full traceback, at least up to the point where your application code triggers a method in `google-cloud-logging`?

**Comment by mishedone** [comment]
Unfortunately the default setup_logging() is using the `google.cloud.logging.handlers.transports.BackgroundThreadTransport` which (at least as I understand it) is triggering an exception in a separate thread (which is good because the main thread is executed successfully - no 500 error). This leaves me with the 3 messages above only taken from StackDriver logs.

On the other hand I've tried something else which may point where the problem is:

```
from google.cloud import logging
from google.cloud.logging.handlers.handlers import CloudLoggingHandler, setup_logging
from google.cloud.logging.handlers.transports import SyncTransport
client = logging.Client()
handler = CloudLoggingHandler(client, transport=SyncTransport)
setup_logging(handler)
```

Basically using the SyncTransport. This one actually worked out pretty well. The following messages got logged successfully without any problem (the second with a stack trace too - neat!):

```
import logging
logger = logging.getLogger('records')
logger.exception('First try.')
try:
    1 / 0
except ZeroDivisionError as e:
    logging.exception('Devision by zero.')
```

So it appears that there is some issue with the BackgroundThreadTransport... I hope this helps.

**Comment by mishedone** [comment]
@tseaver can I provide more information here so we can try finding the issue?

**Comment by mishedone** [comment]
Oh... there's a new version of google-cloud-logging in pip (v1.8.0 - uploaded 4 days ago) - I've just installed it and all went well!

I will close this one since it's solved by upgrading to the latest library version.

**Comment by tseaver** [comment]
@mishedone I'm glad upgrading solved the issue you were having.
